api/api.go

```diff
@@ -1,20 +1,72 @@
 package api
 
-import "fmt"
+import (
+	"encoding/json"
+	"fmt"
+	"io/ioutil"
+	"net/http"
+	"strings"
+)
+
+const splitioAPIUri = "https://sdk.split.io/api"
 
 // SplitioAPIBinding contains splitioAPIKey
 type SplitioAPIBinding struct {
 	splitioAPIKey string
+	splitioAPIUri string
 }
 
 // NewSplitioAPIBinding returns a new SplitioAPIBinding
-func NewSplitioAPIBinding(splitioAPIKey string) *SplitioAPIBinding {
-	return &SplitioAPIBinding{splitioAPIKey}
+func NewSplitioAPIBinding(apiKey string, apiURL string) *SplitioAPIBinding {
+	if apiURL == "" {
+		apiURL = splitioAPIUri
+	}
+	return &SplitioAPIBinding{apiKey, apiURL}
 }
 
-// HTTPGet will make a GET request to the Split.io SDK API
-func (binding *SplitioAPIBinding) HTTPGet() error {
-	return fmt.Errorf("not implemented")
+// HTTPGet makes a GET request to the Split.io SDK API.
+// path is the path of the HTTP request e.g "/splitChanges"
+// since is an integer used as a query string, will be -1 on the first request
+func (binding *SplitioAPIBinding) httpGet(path string, since int) (map[string]interface{}, error) {
+	// initiate a new request
+	req, err := http.NewRequest("GET", fmt.Sprintf("%s%s", binding.splitioAPIUri, path), nil)
+	if err != nil {
+		err := fmt.Errorf("Errored when initiate http new request. %s", err)
+		return map[string]interface{}{}, err
+	}
+
+	// add headers
+	req.Header.Set("Accept", "application/json")
```

Comment:
Worth using a higher-level client (*e.g.*, https://github.com/go-resty/resty)?

Reply:
Thanks for the suggestion. yeah it becomes much cleaner.